File: 1100/1780B/main.cpp
#include<bits/stdc++.h>
using namespace std;

// Euclidean algorithm for GCD
long long gcd(long long a, long long b) 
{
    if (b == 0)
        return a;

    return gcd(b, a % b);
}

int main() 
{
    int t;
    cin>>t;

    while(t--) 
    {
        long long n;
        cin>>n;
        
        vector<long long> arr(n);
        long long sum = 0;

        for(int i = 0; i < n; i++) 
        {
            cin>>arr[i];
            sum += arr[i];
        }

        long long num = 0;
        long long max_gcd = -1;

        // Iterate through all possible prefix partitions (from 0 to n-2)
        for(int i = 0; i < n - 1; i++) 
        {
            num += arr[i];
            max_gcd = max(max_gcd, gcd(sum - num, num));
        }

        cout<<max_gcd<<"\n";
    }

    return 0;
}

File: 1100/1780B/main.java
import java.util.*;
public class main
{
	public static long gcd(long a, long b) 
    {
		if(b == 0) 
            return a;

		return gcd(b,a%b);
	}
 
	public static void main(String args[]) 
    {
        Scanner sc = new Scanner(System.in);
		long t = sc.nextLong();
 
		while(t-- > 0)
        {
		    long n = sc.nextLong();
			long arr[] = new long[(int)n];
 
			long sum = 0, num = 0;
			long max = Long.MIN_VALUE;
 
			for(int i=0; i<n; i++) 
            {
				arr[i] = sc.nextLong();
				sum += arr[i];
			}
 
            for(int i=0; i<n-1; i++) 
            {
				num += arr[i];
				max = Math.max(max, gcd(sum-num,num));
			}
            
            System.out.println(max);
		}
    }
} 